src/Microsoft.AspNet.JsonPatch/Adapters/ObjectAdapter.cs

```diff
@@ -133,91 +126,242 @@ public void Add([NotNull] Operation operation, [NotNull] object objectToApplyTo)
             }
             else
             {
-                positionAsInteger = GetNumericEnd(path);
+                var checkNumericEndResult = GetNumericEndResult(path);
 
-                if (positionAsInteger > -1)
+                if (checkNumericEndResult.HasNumericEnd)
                 {
-                    actualPathToProperty = path.Substring(0,
-                        path.LastIndexOf('/' + positionAsInteger.ToString()));
+                    positionAsInteger = checkNumericEndResult.NumericEnd;
+                    if (positionAsInteger > -1)
+                    {
+                        actualPathToProperty = path.Substring(0,
+                       path.LastIndexOf('/' + positionAsInteger.ToString()));
```

Comment:
Put the `0` and the `path.LastIndexOf(...)` each on their own line so we can see that they are parameters.